### PR TITLE
Refactor and fix block-transfer partitioner

### PR DESCRIPTION
**Pull request by ggezer**

### CODE REVIEW

File: core/server/worker/src/main/java/alluxio/worker/block/management/ManagementTaskCoordinator.java [hunk 1]
```diff
@@ -187,6 +187,7 @@ private void runManagement() {
               e);
         }
         LOG.debug("Management task finished: {}", currentTask.getClass().getSimpleName());
+        Thread.sleep(mLoadDetectionCoolDownMs);
```
Comment (ggezer) [inline core/server/worker/src/main/java/alluxio/worker/block/management/ManagementTaskCoordinator.java:190]: Remove this.

File: core/server/worker/src/main/java/alluxio/worker/block/management/BlockTransferPartitioner.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    Set<BlockStoreLocation> dstLocations = new HashSet<>();
+    // Iterate and process all transfers.
+    for (BlockTransferInfo transferInfo : transferInfos) {
+      if (transferInfo.getSrcLocation().dir() != BlockStoreLocation.ANY_DIR) {
+        srcAllocatedCount++;
+      }
+      if (transferInfo.getDstLocation().dir() != BlockStoreLocation.ANY_DIR) {
+        dstAllocatedCount++;
+      }
+      srcLocations.add(transferInfo.getSrcLocation());
```
[the remark this answers is not in the document]
Comment (ggezer) [reply core/server/worker/src/main/java/alluxio/worker/block/management/BlockTransferPartitioner.java:155]: No, ANY_DIR means unallocated in this context, directory being the allocation unit. If a directory is not known, then it's not right to account it for dir level optimizations.

File: core/server/worker/src/main/java/alluxio/worker/block/management/BlockTransferPartitioner.java [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+      dstLocations.add(transferInfo.getDstLocation());
+    }
+
+    // Find the desired partitioning key.
+    if (srcAllocatedCount == dstAllocatedCount) { // All locations are fully identified.
```
[the remark this answers is not in the document]
Comment (ggezer) [reply core/server/worker/src/main/java/alluxio/worker/block/management/BlockTransferPartitioner.java:160]: Refactored the whole flow.

**Comment by ggezer** [comment]
jenkins, test this please

File: core/server/worker/src/main/java/alluxio/worker/block/management/BlockTransferPartitioner.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (calvinjia) [reply core/server/worker/src/main/java/alluxio/worker/block/management/BlockTransferPartitioner.java:156]: I meant if we should be adding `ANY_DIR` to `src/dstLocations`?

**Comment by ggezer** [comment]
@calvinjia ping!

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Pull request merged**